Feature: Reminding Users to Rate

sagec108 opened the pull request:
Resolves #548 
Implement new pop-up message to remind users to rate severity.

Applies after a fixed number of failures (4) to rate severity.
A failure to rate severity is any instance when the user places a label and closes the context menu without picking a severity.

Closing the context menu is done through:

- clicking outside of the context menu
- clicking the x button
- clicking the OK button
- pressing the enter key

(might also include escape key?)

Minor text fixes.

jonfroehlich commented:
I'd like to talk about the design of this reminder: both the visual design
and the text used in the message itself.

On Tue, May 23, 2017 at 11:42 AM, Sage Chen <notifications@github.com>
wrote:

> Resolves #548
> <https://github.com/ProjectSidewalk/SidewalkWebpage/issues/548>
> Implement new pop-up message to remind users to rate severity.
>
> Applies after a fixed number of failures (4) to rate severity.
> A failure to rate severity is any instance when the user places a label
> and closes the context menu without picking a severity.
>
> Closing the context menu is done through:
>
>    - clicking outside of the context menu
>    - clicking the x button
>    - clicking the OK button
>
> Minor text fixes.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606
> Commit Summary
>
>    - Preliminary Popup Message
>    - Update to popup
>    - Merge remote-tracking branch 'refs/remotes/origin/master' into
>    548-remind-users-to-rate
>    - Removed extraneous code
>    - Merge remote-tracking branch 'refs/remotes/origin/master' into
>    548-remind-users-to-rate
>
> File Changes
>
>    - *M* public/javascripts/SVLabel/src/SVLabel/Main.js
>    <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606/files#diff-0>
>    (2)
>    - *A* public/javascripts/SVLabel/src/SVLabel/alert/
>    RatingReminderAlert.js
>    <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606/files#diff-1>
>    (28)
>    - *M* public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js
>    <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606/files#diff-2>
>    (31)
>    - *M* public/javascripts/index.js
>    <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606/files#diff-3>
>    (4)
>
> Patch Links:
>
>    - https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606.patch
>    - https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/606>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABi-9RFPmZj5a8Nav5dBC_17PIw8A8Usks5r8v5wgaJpZM4Nj5Tg>
> .
>



-- 
Jon Froehlich
Assistant Professor
Computer Science
University of Maryland, College Park
http://www.cs.umd.edu/~jonf/
@jonfroehlich <https://twitter.com/jonfroehlich> - Twitter

sagec108 commented:
I centered the pop-up message and edited the text. Feel free to give input!
![image](https://cloud.githubusercontent.com/assets/20068009/26422442/e4bcc01e-4097-11e7-9005-a95c39f1e4cf.png)

misaugstad commented:
I don't think that "area accessibility" or "location accessibility" would make sense.

As for some (slightly) short options, how about...
Please rate the accessibility of areas you label by pressing keys 1 through 5.
Please rate the accessibility of things you label by pressing keys 1 through 5.
Please rate the accessibility of your labels by pressing keys 1 through 5.

manaswis commented:
Or "Please rate as you apply labels by pressing keys 1 through 5."
Or "Don't forget to rate as you apply labels by pressing keys 1 through 5."

misaugstad commented:
I like the first one that @manaswis mentioned. "Please rate as you apply labels by pressing keys 1 through 5"

jonfroehlich commented:
How about:
- Don't forget to provide severity ratings for each label by pressing keys 1 through 5.
- Please provide severity ratings for each label by pressing keys 1 through 5.

Also, while I like that we are using @Soben713's pop-up message system, I do think that having a message pop-up with the rating context menu would be easier for the user to understand (e.g., the pop-up could have an arrow pointing to the rating menu along with the textual prompt).

manaswis commented:
@sagec108 I tested out the PR. Two comments:

1. Remove the 'period (.)' at the end of the sentence.
2. Currently, when you update the existing labels with a rating value, you will still the get the pop-up. This case is probably not accounted for.
3. Consider a case where you are adding the fourth label and you haven't yet closed the menu, the rating still shows up. I think it should wait the pop-up to close before the pop-up. HOWEVER, this doesn't happen consistently. Would have to test it further. You could also check for it.

sagec108 commented:
I implemented the changes you requested and fixed cases 2 and 3. Updating existing labels should no longer trigger the pop-up and the pop-up should only appear after the menu closes.

manaswis commented on public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js:
```diff
@@ -77,13 +96,32 @@ function ContextMenu (uiContextMenu) {
+        if (labels.length == 0){
+            labels = prev_labels;
+        }
+        if (labels.length > 0) {
```
Is there a specific reason for not going for else if?

jonfroehlich replied:
Did you get a response to this?

Pinging @sagec108

sagec108 replied:
The reason was that the previous labels could also have a 0 length, which is accounted for by the second if statement.
